compiler/env/OMRCPU.cpp

```diff
@@ -29,6 +29,25 @@
 #include "env/defines.h"
 #include "omrcfg.h"
 
+OMR::CPU() :
+   _processor(TR_NullProcessor),
+   _endianness(TR::endian_unknown),
+   _majorArch(TR::arch_unknown),
+   _minorArch(TR::m_arch_none)
+   {
+   _processorDescription.processor = OMR_PROCESSOR_UNDEFINED;
+   }
+
+OMR::CPU(const OMRProcessorDesc& processorDescription) :
+   _processorDescription(processorDescription)
+   {}
+
+OMR::CPU(OMRPortLibrary* omrPortLib)
+   {
+   OMRPORT_ACCESS_FROM_OMRPORT(omrPortLib);
+   omrsysinfo_get_processor_description(&_processorDescription);
+   }
```

Comment:
I'd probably not even have this constructor given how simple it is to extract an `OMRProcessorDesc` structure from the port library. It's not like users would have more than one `OMRPortLibrary*` anyway. That also reduces the cost to downstream projects who extend this class.